frontend/src/metabase/visualizations/echarts/cartesian/option/ticks.ts

```diff
@@ -0,0 +1,114 @@
+import type { Dayjs } from "dayjs";
+import dayjs from "dayjs";
+import type { ContinuousDomain } from "metabase/visualizations/shared/types/scale";
+import type {
+  TimeSeriesAxisFormatter,
+  TimeSeriesXAxisModel,
+} from "../model/types";
+import {
+  computeTimeseriesTicksInterval,
+  getLargestInterval,
+  getTimeSeriesIntervalDuration,
+} from "../utils/timeseries";
+
+// HACK: ECharts in some cases do not render two ticks on line charts with 1 interval (2 values) when minInterval is defined.
+// For example, when a dataset has two days and minInterval is 1 day in milliseconds datasets like ["2022-01-01", "2022-01-02"]
+// will be rendered without the second tick. However, for ["2022-01-02", "2022-01-03"] ECharts would corectly render two ticks as needed.
+// The workaround is to add more padding on sides for this corner case.
+const getPadding = (intervalsCount: number) => {
+  if (intervalsCount === 1) {
+    return 5 / 6;
+  }
+
+  return 0.5;
+};
+
+export const getTicksOptions = (
+  xAxisModel: TimeSeriesXAxisModel,
+  chartWidth: number,
+) => {
+  const { range } = xAxisModel;
+
+  let formatter: TimeSeriesAxisFormatter = xAxisModel.formatter;
+  let minInterval: number | undefined;
+  let maxInterval: number | undefined;
+
+  const dataInterval = xAxisModel.interval;
```

Comment:
Nit: we use both `dataInterval` and `xAxisModel.interval` in this function even though they are the same (`dataInterval` is never mutated). We should just use one or the other.